where-to-travel/src/main/webapp/script.js

```diff
@@ -447,46 +502,72 @@ function populatePlaces(placeArray) {
       focusedCard.scrollIntoView({behavior: 'smooth', block: 'center'});
     });
 
-    placeMarker.addListener('mouseover', function () {
-      placeMarker.setIcon(SELECTED_PIN_PATH);
-    });
+      placeMarker.addListener('mouseover', function () {
+        placeMarker.setIcon(SELECTED_PIN_PATH);
+      });
 
-    placeMarker.addListener('mouseout', function () {
-      if (placeMarker != focusedPin) {
-        placeMarker.setIcon(PIN_PATH);
-      }
-    });
+      placeMarker.addListener('mouseout', function () {
+        if (placeMarker != focusedPin) {
+          placeMarker.setIcon(PIN_PATH);
+        }
+      });
 
-    markers.push(placeMarker);
+      markers.push(placeMarker);
+    }
   }
 
   document.getElementById(SCROLL_ID).hidden = false;
-
   // Handle favoriting a place
   $('.icon').click(function() {
+    const name = $(this).parent().parent().parent().attr('placeName');
+    const placeId = $(this).parent().next().next().next().attr('savedPlaceId');
+    let card = document.getElementById(name);
     $(this).toggleClass('press');
     if (firebase.auth().currentUser && $(this).hasClass('press')) {
-      const name = $(this).parent().parent().parent().attr('placeName');
-      const link = $(this).parent().next().attr('href');
       const time = $(this).parent().next().next().text();
        // Add users saved places to the real time database in Firebase when star is pressed
       const database = firebase.database();
       var uID = firebase.auth().currentUser.uid;
-      var ref = database.ref('users/' + uID + '/' + 'places' + '/' + name);
+      var ref = database.ref('users/' + uID + '/places/' + name);
       var data = {
         name: name,
-        link: link,
-        time: time,
+        timeAsString: time,
+        timeInSeconds: card.getAttribute('data-timeInSeconds'),
+        place_id: placeId,
       }
       ref.set(data);
+      let lat = card.dataset.lat;
```

Comment:
Add a comment to this block similar to above?

Reply:
@emmapaczkowski could you do this one?

Reply:
done